src/stats.rs

```diff
@@ -0,0 +1,160 @@
+        self.0.calls_count.fetch_add(1, Ordering::Relaxed);
+    }
+
+    pub fn track_completed_call(&self, duration: Duration) {
+        self.0.calls_completed_count.fetch_add(1, Ordering::Relaxed);
+        self.0.milliseconds_waited.fetch_add(
+            usize::try_from(duration.as_millis()).unwrap_or(usize::MAX),
+            Ordering::Relaxed,
+        );
+    }
+
+    pub async fn run_in_background(self) {
+        let mut interval = tokio::time::interval(Duration::from_secs(1));
+        interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
+
+        let mut last_logged = Instant::now();
+        let log_interval = Duration::from_secs(60);
+        let mut shutdown_signal = self.0.shutdown_signal.signal();
+
+        loop {
+            let calls_count_before = self.0.calls_count.load(Ordering::Relaxed);
+            let start = Instant::now();
+
+            tokio::select! {
+                _ = &mut shutdown_signal => {
+                    break
+                }
+                _ = interval.tick() => {}
+            };
+
+            let mut summary = self.0.summary.lock().await;
+            let elapsed = start.elapsed().as_secs_f64();
+
+            // update highest TPS
+            let calls_count_after = self.0.calls_count.load(Ordering::Relaxed);
+            let calls_count =
+                u32::try_from(calls_count_after - calls_count_before).unwrap_or(u32::MAX);
+            let tps = if elapsed > 0.0 {
+                f64::from(calls_count) / elapsed
+            } else {
+                0.0
+            };
+            if tps > summary.tps_highest {
+                summary.tps_highest = tps;
+            }
+
+            // update average TPS
+            let elapsed_total = summary.start.elapsed().as_secs_f64();
+            let calls_count_total = u32::try_from(calls_count_after).unwrap_or(u32::MAX);
+            summary.tps_average = if elapsed_total > 0.0 {
+                f64::from(calls_count_total) / elapsed_total
+            } else {
+                0.0
+            };
+
+            // update time spent waiting for MSE calls to complete
+            summary.wait_time_total = Duration::from_millis(
+                u64::try_from(self.0.milliseconds_waited.load(Ordering::Relaxed))
+                    .unwrap_or(u64::MAX),
+            );
+            summary.wait_time_average = summary
+                .wait_time_total
+                .checked_div(calls_count_total)
+                .unwrap_or_default();
+
+            // update pending requests
+            summary.pending_current =
+                calls_count_after - self.0.calls_completed_count.load(Ordering::Relaxed);
+            if summary.pending_current > summary.pending_highest {
+                summary.pending_highest = summary.pending_current;
+            }
+
+            // log summary every minute
+            if last_logged.elapsed() > log_interval {
+                last_logged = Instant::now();
+
+                log::info!(
```

Comment:
Having run the tests can we get the performance logs to also be put in the `dcs.log` ? I didn't see them during my testing. It will help server admins having it inline with everything else.